umi_tools/umi_methods.py

```diff
@@ -45,7 +45,9 @@ def get_umi_tag(read, tag='RX'):
     ''' extract the umi from the specified tag '''
 
     try:
-        return read.get_tag(tag).encode('utf-8')
+        # 10X pipelines append a 'GEM' tag to the UMI, e.g
+        # AGAGSGATAGATA-1
+        return read.get_tag(tag).split("-")[0].encode('utf-8')
```

Comment:
Profiled master vs {TS}-RetainUnmapped using the test input file for extraction from bam 
e.g:
`python -m cProfile -o profile_master.cprof ../umi_tools/dedup.py --stdin=chr19_tag.bam -L test_master.log --out-sam --random-seed=123456789 --method=directional --umi-tag=RX --extract-umi-method=tag > test.bam`

5 times for each branch. Total times for 55197 calls to umi_methods.get_umi_tag:
Master: 0.027-0.029s
{TS}-RetainUnmapped: 0.043-0.051s

For reference, the total run time was 2.1-2.3s and the 55197 calls to pysam.get_tag took 0.83-0.100. 

umi_get_tag is only ever going to be called a maximum of 2*Number of reads. For input with lots of reads but simple networks, this will add <2% to the run time. For input with complex networks, the impact on run time will be minimal.